BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -430,33 +451,45 @@ public class SwapTokenStore: ObservableObject {
   func prepare(with accountInfo: BraveWallet.AccountInfo, completion: (() -> Void)? = nil) {
     self.accountInfo = accountInfo
     
-    tokenRegistry.allTokens { [self] tokens in
-      let fullList = tokens + [.eth]
-      allTokens = fullList.sorted(by: { $0.symbol < $1.symbol })
-      
+    func updateSelectedTokens(chainId: String) {
       if let fromToken = selectedFromToken { // refresh balance
-        rpcController.balance(for: fromToken, in: accountInfo) { balance in
-          if let value = balance {
-            selectedFromTokenBalance = BDouble(value)
-          }
+        rpcController.balance(for: fromToken, in: accountInfo) { [weak self] balance in
+          self?.selectedFromTokenBalance = BDouble(balance ?? 0)
         }
       } else {
         selectedFromToken = allTokens.first(where: { $0.isETH })
       }
       
-      rpcController.chainId { [self] chainId in
-        if let toToken = selectedToToken {
-          rpcController.balance(for: toToken, in: accountInfo) { balance in
-            selectedToTokenBalance = BDouble(balance ?? 0)
-            completion?()
-          }
+      if let toToken = selectedToToken {
+        rpcController.balance(for: toToken, in: accountInfo) { [weak self] balance in
+          self?.selectedToTokenBalance = BDouble(balance ?? 0)
+          completion?()
+        }
+      } else {
+        if chainId == BraveWallet.MainnetChainId {
+          selectedToToken = allTokens.first(where: { $0.symbol == "BAT" })
+        } else if chainId == BraveWallet.RopstenChainId {
+          selectedToToken = allTokens.first(where: { $0.symbol == "DAI" })
```

Comment:
Do a case insensitive compare or uppercase `$0.symbol` to ensure a normalized compare